Problems building Ambassador

I tried building Ambassador from source. 

The tests were failing for me because of errors in the dependencies Scout and peewee. Here were the errors I was getting

```
tests/ambassador_test.py:13: in <module>
    from diag_paranoia import diag_paranoia, filtered_overview
tests/diag_paranoia.py:9: in <module>
    from ambassador.config import Config
ambassador/config.py:37: in <module>
    class Config (object):
ambassador/config.py:82: in Config
    _scout_args['id_plugin'] = Scout.configmap_install_id_plugin
E   AttributeError: type object 'Scout' has no attribute 'configmap_install_id_plugin'
```
With peewee I ran into this [problem](https://github.com/coleifer/peewee/issues/1218).

I was able to get passed the peewee error by modifying the source but I gave up on the Scout issues.

Reporting this here in case others try to build it and run into the same problem.

I was trying to build from source because I want to use a custom envoy.j2 template and based on #207 it looks like overriding the config might not be well supported. So I figured I'd just modify source and rebuild.

Due to build problems above I'm going to just try creating a new image using the 0.22.0 image as a base image and just overriding the envoy.j2 file.

--- a/.gitignore
+++ b/.gitignore
@@ -34,6 +34,9 @@ tags
 no-such-file
 envoy.json
 intermediate.json
+check-*.json
 end-to-end/ambassador-deployment-mounts.yaml
 end-to-end/ambassador-deployment.yaml
+end-to-end/005-single-namespace/k8s/ambassador-deployment.yaml
+end-to-end/kubernaut
 end-to-end/*.log

--- a/dev-requirements.txt
+++ b/dev-requirements.txt
@@ -5,4 +5,3 @@ awscli
 pytest
 pytest-cov
 dpath
-kubernaut

--- a/end-to-end/README.md
+++ b/end-to-end/README.md
@@ -2,6 +2,7 @@ Run `testall.sh` in this directory to descend into each 0* directory and run the
 
 TESTS REQUIRE:
 
-kubernaut
 Python 3
 everything in dev-requirements.txt
+
+Note that running the end-to-end tests will install [kubernaut](kubernaut.io) as `end-to-end/kubernaut`.

--- a/end-to-end/README.md
+++ b/end-to-end/README.md
@@ -1,10 +1,8 @@
 Run `testall.sh` in this directory to descend into each 0* directory and run the tests.
 
-#### Requires:
+- You will need to use Python 3 to run the tests.
+- The tests use [`kubernaut`](https://kubernaut.io/). If needed, they will install `kubernaut` for you and walk you through getting a `kubernaut` token.
+- The tests will not work unless you've pushed your build to a public `DOCKER_REGISTRY`. See [building instructions](../BUILDING.md).
 
-* Python 3
-* [kubernaut](https://github.com/datawire/kubernaut#installation)
-* kubernaut token from https://kubernaut.io/token
-* A build pushed to a public DOCKER_REGISTRY. See [building instructions](../BUILDING.md).
+If you're working with this stuff, we'd love to see you in our [Gitter channel](https://gitter.im/datawire/ambassador)!
 
-Note that running the end-to-end tests will install [kubernaut](kubernaut.io) as `end-to-end/kubernaut`.

--- a/end-to-end/utils.sh
+++ b/end-to-end/utils.sh
@@ -84,6 +84,7 @@ wait_for_ready () {
 
     if [ -z "$ready" ]; then
         echo 'Ambassador not yet ready?' >&2
+        kubectl get pods >&2
         exit 1
     fi
 }

--- a/end-to-end/utils.sh
+++ b/end-to-end/utils.sh
@@ -13,13 +13,43 @@ step () {
     echo "==== $@"
 }
 
-shred_and_reclaim () {
+get_kubernaut () {
     KUBERNAUT="$ROOT/kubernaut"
 
     if [ ! -x "$KUBERNAUT" ]; then
         echo "Fetching Kubernaut..."
         curl -q -L -o "$KUBERNAUT" https://s3.amazonaws.com/datawire-static-files/kubernaut/$(curl -s https://s3.amazonaws.com/datawire-static-files/kubernaut/stable.txt)/kubernaut
         chmod +x "$KUBERNAUT"
+    fi
+
+    echo "$KUBERNAUT"
+}
+
+check_kubernaut_token () {
+    KUBERNAUT="$1"
+    
+    if [ $("$KUBERNAUT" kubeconfig | grep -c 'Token not found') -gt 0 ]; then
+        echo "You need a Kubernaut token. Go to"
+        echo ""
+        echo "https://kubernaut.io/token"
+        echo ""
+        echo "to get one, then run"
+        echo ""
+        echo "sh $ROOT/save-token \"\$token\""
+        echo ""
+        echo "to save it before trying again."
+
+        return 1
+    fi
+
+    return 0
+}
+
+shred_and_reclaim () {
+    KUBERNAUT=$(get_kubernaut)
+
+    if ! check_kubernaut_token "$KUBERNAUT"; then
+        exit 1
     fi
 
     step "Dropping old cluster"

--- a/end-to-end/utils.sh
+++ b/end-to-end/utils.sh
@@ -1,3 +1,8 @@
+if [ -z "$ROOT" ]; then
+    echo "ROOT must be set to the root of the end-to-end tests" >&2
+    exit 1
+fi
+
 if [ -n "$MACHINE_READABLE" ]; then
     LINE_END="\n"
 else
@@ -9,11 +14,19 @@ step () {
 }
 
 shred_and_reclaim () {
+    KUBERNAUT="$ROOT/kubernaut"
+
+    if [ ! -x "$KUBERNAUT" ]; then
+        echo "Fetching Kubernaut..."
+        curl -q -L -o "$KUBERNAUT" https://s3.amazonaws.com/datawire-static-files/kubernaut/$(curl -s https://s3.amazonaws.com/datawire-static-files/kubernaut/stable.txt)/kubernaut
+        chmod +x "$KUBERNAUT"
+    fi
+
     step "Dropping old cluster"
-    kubernaut discard
+    "$KUBERNAUT" discard
 
     step "Claiming new cluster"
-    kubernaut claim
+    "$KUBERNAUT" claim
     export KUBECONFIG=${HOME}/.kube/kubernaut
 }
 

